### DISCUSSION TITLE
Limit collection generation when working locally

### DISCUSSION THREAD

**Discussion by msm1227** [11ty/eleventy#3025] [Q&A] [answered]
Hey all,

I am wondering if there is a way to limit the amount of posts that are generated to the _site folder when working locally. I am working on a site that has 400+ blog posts and it takes a while for my site to reload after making a change locally. If I move all put ~50 of those to a temp folder I am able to work a bit faster since everything reloads quicker. Is there a way to limit this without me having to manually move the posts between commits to my repo?

Thanks!

**Comment by cfjedimaster** [comment]
I'd use `.eleventyignore`. I use it on my site (6k+ blog posts) to hide like 90% of the content.

**Reply by pdehaan** [reply, marked as answer]
@msm1227 there is an excellent blog post (by @cfjedimaster) about it too at https://www.raymondcamden.com/2021/10/15/eleventy-10-dynamic-ignores

**Reply by cfjedimaster** [reply]
I honestly forgot I wrote that.

**Reply by pdehaan** [reply]
@cfjedimaster You should check out your blog some time, it's pretty good. 10/10, would read again. 🎉

**Reply by pdehaan** [reply]
I also tried using this, but it just seems to grab the 20 most alphabetical posts if they're all in a flat directory structure, so not sure if it's useful at all:
```js
const fg = require("fast-glob");

/**
 * @param {import("@11ty/eleventy/src/UserConfig")} eleventyConfig
 * @returns {ReturnType<import("@11ty/eleventy/src/defaultConfig")>}
 */
module.exports = function (eleventyConfig) {
  if (process.env.ELEVENTY_ENV === "development") {
    fg.sync("./src/pages/**/*.njk")
      .splice(20)
      .forEach(p => eleventyConfig.ignores.add(p));
  }

  eleventyConfig.setQuietMode(true);

  return {
    dir: {
      input: "src",
      output: "www",
    }
  };
};
```

```sh
# Build everything:
> eleventy
[11ty] Wrote 201 files in 0.14 seconds (0.7ms each, v2.0.1)

# Run in "development" mode and limit it to 20 ./src/pages/*.njk templates (while building everything else):
> ELEVENTY_ENV=development eleventy
[11ty] Wrote 21 files in 0.06 seconds (2.9ms each, v2.0.1)
```

Wasn't sure how I could say "build the most recent _n_ posts" without needing to do a bunch of disk i/o to get timestamps for each file. This seems to work, but feels like excessive disk operations:

```js
const fs = require("node:fs");
const fg = require("fast-glob");

/**
 * @param {import("@11ty/eleventy/src/UserConfig")} eleventyConfig
 * @returns {ReturnType<import("@11ty/eleventy/src/defaultConfig")>}
 */
module.exports = function (eleventyConfig) {
  if (process.env.ELEVENTY_ENV === "development") {
    fg.sync("./src/pages/**/*.njk")
      .sort((a, b) => fs.statSync(b).birthtimeMs - fs.statSync(a).birthtimeMs)
      .splice(process.env.DRAFT_COUNT ?? 20)
      .forEach(p => eleventyConfig.ignores.add(p));
  }

  eleventyConfig.setQuietMode(process.env.ELEVENTY_ENV !== "development");

  return {
    dir: {
      input: "src",
      output: "www",
    }
  };
};
```

```sh
# Build everything:
> eleventy
[11ty] Wrote 202 files in 0.10 seconds (0.5ms each, v2.0.1)

# Build `n` most recent posts (default is 20):
> ELEVENTY_ENV=development DRAFT_COUNT=4 eleventy
[11ty] Writing www/index.html from ./src/index.njk
[11ty] Writing www/pages/page-198/index.html from ./src/pages/page-198.njk
[11ty] Writing www/pages/page-199/index.html from ./src/pages/page-199.njk
[11ty] Writing www/pages/page-200/index.html from ./src/pages/page-200.njk
[11ty] Writing www/pages/page-2023/index.html from ./src/pages/page-2023.njk
[11ty] Wrote 5 files in 0.05 seconds (v2.0.1)
```

**Reply by pdehaan** [reply]
Turns out fast-glob has a [`stats` option](https://www.npmjs.com/package/fast-glob#stats) which means I don't need to call `fs.statSync()`:

```js
  if (process.env.ELEVENTY_ENV === "development") {
    fg.sync("./src/pages/**/*.njk", { stats: true })
      // Sort by descending date (newest first).
      .sort((a, b) => b.stats.birthtimeMs - a.stats.birthtimeMs)
      .splice(process.env.DRAFT_COUNT ?? 20)
      .forEach(p => eleventyConfig.ignores.add(p.path));
  }
```

**Reply by msm1227** [reply]
Thanks so much for the help everyone! I went the route that [cfjedimaster](https://github.com/cfjedimaster) mentioned in his blog post. Also I've come across your stuff in the past, thanks for the awesome work!
